src/main/java/org/kohsuke/github/GHOrganization.java

```diff
@@ -449,11 +449,38 @@ public GHProject createProject(String name, String body) throws IOException {
 
     /**
      * The enum Permission.
+     *
+     * @deprecated Use <code>RolePermission</code> instead, which allows for custom permissions/roles created in github
+     *             to be set.
      */
+    @Deprecated
     public enum Permission {
         ADMIN, MAINTAIN, PUSH, TRIAGE, PULL
     }
 
+    /**
+     * Permissions (roles) for teams and collaborators.
+     */
+    public static class RolePermission {
+        // pre-canned permissions/roles
+        public static final RolePermission ADMIN = new RolePermission("admin");
+        public static final RolePermission MAINTAIN = new RolePermission("maintain");
+        public static final RolePermission PULL = new RolePermission("pull");
+        public static final RolePermission PUSH = new RolePermission("push");
+        public static final RolePermission TRIAGE = new RolePermission("triage");
+
+        private final String permission;
+
+        public RolePermission(String permission) {
+            this.permission = permission;
+        }
```

Comment:
Make it easy for people to understand how this works. 

```suggestion
        private RolePermission(String permission) {
            this.permission = permission;
        }
        
        public RolePermission custom(String permission) {
            return new RolePermission(permission);
        }
        
        RolePermission from(Permission permission) {
            return new RolePermission(permission);
        }
```